### PR TITLE
Optionally loosen `requireParam` when implements/augments/extends present on tag or parent

### PR DESCRIPTION
**Pull request by brettz9**
I wasn't sure whether to add them as settings rather than rule-specific options, but I followed your pattern.

### CODE REVIEW

File: src/iterateJsdoc.js [hunk 1]
```diff
@@ -100,7 +134,11 @@ export default (iterator) => {
         }
       };
 
-      const utils = curryUtils(functionNode, jsdoc, tagNamePreference, additionalTagNames, allowOverrideWithoutParam);
+      const utils = curryUtils(
+        functionNode, jsdoc, tagNamePreference, additionalTagNames,
```
Comment (gajus) [inline src/iterateJsdoc.js]: style: Please use new line for every parameter.

File: src/rules/requireParam.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import _ from 'lodash';
 import iterateJsdoc from '../iterateJsdoc';
 
-export default iterateJsdoc(({
+export default iterateJsdoc(({ // eslint-disable-line complexity
```
Comment (gajus) [inline src/rules/requireParam.js]: style: Please use `// eslint-disable-next-line  complexity`

File: src/iterateJsdoc.js [hunk 3]
```diff
@@ -33,41 +56,52 @@ const curryUtils = (functionNode, jsdoc, tagNamePreference, additionalTagNames,
...
+    const allowImplementsWithoutParam = Boolean(_.get(context, 'settings.jsdoc.allowImplementsWithoutParam'));
+    const allowAugmentsExtendsWithoutParam = Boolean(_.get(context, 'settings.jsdoc.allowAugmentsExtendsWithoutParam'));
 
     const checkJsdoc = (functionNode) => {
       const jsdocNode = sourceCode.getJSDocComment(functionNode);
 
       if (!jsdocNode) {
         return;
       }
+      const ancestors = context.getAncestors();
```
Comment (gajus) [inline src/iterateJsdoc.js]: style: add a space before ancestors.

File: src/iterateJsdoc.js [hunk 4]
```diff
@@ -33,41 +56,52 @@ const curryUtils = (functionNode, jsdoc, tagNamePreference, additionalTagNames,
...
+  utils.isAugmentsExtendsAllowedWithoutParam = () => {
+    return allowAugmentsExtendsWithoutParam;
+  };
 
-        return commentParser.PARSERS.parse_name(str, data);
-      },
-      commentParser.PARSERS.parse_description
-    ]
-  })[0] || {};
+  utils.classHasTag = function (tagName) {
```
Comment (gajus) [inline src/iterateJsdoc.js]: style: use arrow function.

File: src/iterateJsdoc.js (hunk 3 above)
Comment (brettz9) [reply src/iterateJsdoc.js]: I guess you mean a line break, right? It's at the right indent...

**Comment by brettz9** [comment]
I've amended to address the styling issues...

**Comment by gajus** [comment]
🎉

**Pull request merged**